vonk/reference/programming_api/elementmodel.rst

```diff
@@ -0,0 +1,176 @@
+.. _vonk_reference_api_elementmodel:
+
+ElementModel, manipulating IResource and ISourceNode
+====================================================
+
+As stated in :ref:`vonk_reference_api_iresource`, ``IResource`` is immutable, and so is ``ISourceNode`` that it is based on.
+Still, there are cases where you would want to manipulate a resource. Add, change or remove elements. The namespace ``Vonk.Core.ElementModel`` has methods to do so. All of these methods do *NOT* change the original structure (the input IResource or ISourceNode), but instead *return* an updated structure.
+
+.. _vonk_reference_api_isourcenode:
+
+ISourceNode manipulation
+------------------------
+
+All the ``ISourceNode`` extension methods can be used on ``IResource`` as well. If you need an ``IResource`` as result, just turn the resulting ``ISourceNode`` to an ``IResource`` again. So if you added an element to an ISourceNode:
+
+.. code-block:: csharp
+
+   IResource result = input.AddIfNotExists(SourceNode.Valued("active", "false")).ToIResource(input.InformationModel);
+
+For some of the more common extension methods we provide an overload on IResource that does this for you, like ``IResource.Patch(...)``
+
+All the methods below are in the namespace ``Vonk.Core.ElementModel.ISourceNodeExtensions``
+
+:method: ISourceNode Add(this ISourceNode original, ISourceNode newChild)
```

Comment:
I don't really like the way this looks on the page. Perhaps a table with lines between each method would help, but at this moment I'm not sure how to format this nicer in rst.